packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -467,27 +465,9 @@ export default class SalesforceCRMService implements CRM {
         appOptions?.roundRobinSkipFallbackToLeadOwner &&
         forRoundRobinSkip
       ) {
-        const searchResult = await conn.search(
```

Comment:
This call to check  for a lead was made after the first call to check for a contact. Instead we can combine both calls into one.